actions/pypi_build/action.yaml

```diff
@@ -0,0 +1,56 @@
+name: "Build PyPi"
+description: "Build PyPi wheel"
+
+inputs:
+    dev:
+      description: 'If the build is a dev build'
+      required: false
+      default: false
+    release:
+      description: 'If the build is a release build'
+      required: false
+      default: false
+    name:
+      description: 'Name to append to the wheel'
+      required: false
+     
+outputs:
+    filename:
+      description: 'wheel filename'
+      value: ${{ steps.build.outputs.filename }}
+
+# can we just pass in build args instead of updating the variables in version.py?
+runs:
+  using: "composite"
+  steps:
+    - name: build 
+      shell: bash
+      run:  |
+            pwd
+            sudo apt-get -y install python3-pip
+            pip3 --version
+            sudo pip3 install virtualenv
+            virtualenv venv
+            source venv/bin/activate
+            pip install -e .
+            name="${{ inputs.name }}"
+            if ${{ inputs.release }}; then
+              sed -i 's/is_release = False/is_release = True/g' src/${{ github.event.repository.name }}/version.py
+            elif ${{ inputs.dev }}; then
+              sed -i 's/is_dev = False/is_dev = True/g' src/${{ github.event.repository.name }}/version.py
+              sed -i 's/dev_number = None/dev_number = '"$name"'/g' src/${{ github.event.repository.name }}/version.py
```

Comment:
I can see why you used sed to replace that appropriate part of the code. This is a bit hacky. 
For a simpler fix instead of replacing, use is_release env variable as input?

Reply:
We need the version.py to be updated in order for setup.py to pull the correct version. We could look into passing in build args to setup.py (essentially what I mentioned in my comment in the on line 22.

Reply:
did you look into what QA currently does here?